### PR TITLE
Fix search-icon when typing

### PR DESCRIPTION
**Pull request by Subhajit009iitr**
This PR fixes the `Search-icon` disappearance issue when typing something in the search bar. Earlier the magnifying glass symbol was kept within the placeholder of the input form, so whenever something was typed the `Search-icon` would disappear too along with the placeholder comments `Search this site`. I have fixed that in this PR and now the magnifying glass `Search-icon` stays in place at the left-most side of the search bar. 
This is in the context of the issue #44416.

### CODE REVIEW

**Review by Gauravpadam — COMMENTED** [review]
Thanks @Subhajit009iitr

I've seen the approach and I have a few suggestions to make:

- Try to use an `fa search` (fontawesome search) icon instead of a background png; CSS would then change accordingly
- The internal CSS is OK; Even better if you can accomodate it into `assets/custom.scss`

**Comment by Subhajit009iitr** [comment]
@Gauravpadam I have updated the code using the ``fa search`` icon
The code still has internal CSS which won't cause any functionality problems for now, If you allow I would like to work on it by opening another issue, to isolate the CSS part as it may require changing other files.

**Comment by Subhajit009iitr** [comment]
The search bar looks like this now
image

File: layouts/partials/search-input.html [hunk 1]
```diff
@@ -1,30 +1,82 @@
...
-  <input
-    type="search"
-    class="form-control td-search-input"
-    placeholder="&#xf002; {{ T "ui_search_placeholder" }}"
-    aria-label="{{ T "ui_search_placeholder" }}"
-    autocomplete="off"
-  >
-{{ else if .Site.Params.offlineSearch }}
-  <div id="search-nav-container">
+<head>
```
Comment (sftim) [inline layouts/partials/search-input.html:1]: Why add this element?

Comment (Subhajit009iitr) [reply layouts/partials/search-input.html:1]: It was initially there in the code, I thought it may have some CSS properties attached to it

Comment (Gauravpadam) [inline layouts/partials/search-input.html:1]:
```suggestion
```
This is a partial component, It doesn't need a head

File: layouts/partials/search-input.html [hunk 2]
```diff
@@ -1,30 +1,82 @@
...
+    }
+
+    .search-input::placeholder {
+      font-style: normal;
+      font-weight: normal;
+      font-size: 16px;
+      color: #aaa;
+    }
+  </style>
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0/css/all.min.css">
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:42]:
```suggestion
```
We already have this css integrated in the website

File: layouts/partials/search-input.html [hunk 3]
```diff
@@ -1,30 +1,82 @@
...
+</head>
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:43]:
```suggestion
```

File: layouts/partials/search-input.html [hunk 4]
```diff
@@ -1,30 +1,82 @@
...
+
+<body>
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:45]:
```suggestion
```
Partial component; body not needed

File: layouts/partials/search-input.html [hunk 5]
```diff
@@ -1,30 +1,82 @@
...
-    class="form-control td-search-input"
-    name="q"
-    placeholder="&#xf002 {{ T "ui_search_placeholder" }}"
-    aria-label="{{ T "ui_search_placeholder" }}"
-    autocomplete="off"
-  >
-{{ end }}
\ No newline at end of file
+  {{ end }}
+</body>
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:82]:
```suggestion
```

File: layouts/partials/search-input.html [hunk 6]
```diff
@@ -1,30 +1,82 @@
...
-  >
-{{ else if .Site.Params.offlineSearch }}
-  <div id="search-nav-container">
+<head>
+  <style>
+    .search-bar {
+      display: flex;
+      align-items: center;
+      background-color: #fff;
+      border: 1px solid #ccc;
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:7]:
```suggestion
      border: 1px solid #4c4c4c;
```

File: layouts/partials/search-input.html [hunk 7]
```diff
@@ -1,30 +1,82 @@
...
+      border-radius: 20px; 
+      padding: 4px; 
+      width: 50vw;
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:10]:
```suggestion
      width: auto;
```

File: layouts/partials/search-input.html [hunk 8]
```diff
@@ -1,30 +1,82 @@
...
+      font-size: 16px;
+      border-radius: 15px;
+    }
+
+    .search-input::placeholder {
+      font-style: normal;
+      font-weight: normal;
+      font-size: 16px;
+      color: #aaa;
+    }
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:40]:
```suggestion
```
This doesn't change much, let's omit this block

File: layouts/partials/search-input.html [hunk 9]
```diff
@@ -1,30 +1,82 @@
...
+      width: 50vw;
+    }
+
+    .search-bar:focus-within {
+      border: 2.5px solid rgba(47, 135, 223, 0.7);
+    }
+
+    .search-bar i.fas {
+      margin-right: 10px; 
+    }
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:19]:
```suggestion
    .search-bar i.search-icon {
      padding: .5em .5em .5em .75em;
      opacity: .75;
    }
```
This looks better

File: layouts/partials/search-input.html [hunk 10]
```diff
@@ -1,30 +1,82 @@
...
+
+    .td-search-input {
+      font-family: sans-serif;
+    }
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:23]:
```suggestion
```
Site assets (external css) handles it

File: layouts/partials/search-input.html [hunk 11]
```diff
@@ -1,30 +1,82 @@
...
-  <div id="search-nav-container">
+<head>
+  <style>
+    .search-bar {
+      display: flex;
+      align-items: center;
+      background-color: #fff;
+      border: 1px solid #ccc;
+      border-radius: 20px; 
+      padding: 4px;
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:9]:
```suggestion
      vertical-align: middle;
      flex-grow: 1;
      overflow-x: hidden;
```
Some edge cases I spotted will need these; also, scrap the padding here

File: layouts/partials/search-input.html [hunk 12]
```diff
@@ -1,30 +1,82 @@
...
+      color: #aaa;
+    }
+  </style>
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0/css/all.min.css">
+</head>
+
+<body>
+  {{ if or .Site.Params.gcs_engine_id .Site.Params.algolia_docsearch }}
+  <div class="search-bar">
+    <i class="fas fa-search"></i>
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:48]:
```suggestion
    <i class="search-icon fas fa-search"></i>
```

File: layouts/partials/search-input.html [hunk 13]
```diff
@@ -1,30 +1,82 @@
...
       type="search"
+      class="search-input td-search-input" 
+      placeholder="{{ T "ui_search_placeholder" }}"
+      aria-label="{{ T "ui_search_placeholder" }}"
+      autocomplete="off"
+    >
+  </div>
+  {{ else if .Site.Params.offlineSearch }}
+  <div class="search-bar" id="search-nav-container">
+    <i class="fas fa-search"></i>
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:59]:
```suggestion
    <i class="search-icon fas fa-search"></i>
```

File: layouts/partials/search-input.html [hunk 14]
```diff
@@ -1,30 +1,82 @@
...
+      placeholder="{{ T "ui_search_placeholder" }}"
+      aria-label="{{ T "ui_search_placeholder" }}"
       autocomplete="off"
-      class="form-control td-search-input"
-      placeholder="&#xf002 {{ T "ui_search_placeholder" }}"
+    >
+  </div>
+  {{ else if .Site.Params.k8s_search }}
+  <div class="search-bar">
+    <i class="fas fa-search"></i>
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:71]:
```suggestion
    <i class="search-icon fas fa-search"></i>
```

File: layouts/partials/search-input.html [hunk 15]
```diff
@@ -1,30 +1,82 @@
...
+
+    .search-input {
+      flex: 1;
+      box-sizing: border-box;
+      border: none;
+      outline: none;
+      padding: 8px;
+      font-size: 16px;
+      border-radius: 15px;
+    }
```
Comment (Gauravpadam) [inline layouts/partials/search-input.html:33]:
```suggestion
    .search-input {
      flex: 1;
      border: none;
      outline: none;
      padding: .5em 0 .5em 0;
    }
```

**Review by Gauravpadam — COMMENTED** [review]
Thanks

Please see my inline feedback for some further improvements on the approach

File: layouts/partials/search-input.html (hunk 1 above)
Comment (Subhajit009iitr) [reply layouts/partials/search-input.html]: done

File: layouts/partials/search-input.html (hunk 2 above)
Comment (Subhajit009iitr) [reply layouts/partials/search-input.html]: done

File: layouts/partials/search-input.html (hunk 3 above)
Comment (Subhajit009iitr) [reply layouts/partials/search-input.html]: done

File: layouts/partials/search-input.html (hunk 4 above)
Comment (Subhajit009iitr) [reply layouts/partials/search-input.html]: done

File: layouts/partials/search-input.html (hunk 5 above)
Comment (Subhajit009iitr) [reply layouts/partials/search-input.html]: done

File: layouts/partials/search-input.html (hunk 6 above)
Comment (Subhajit009iitr) [reply layouts/partials/search-input.html]: done

**Comment by Subhajit009iitr** [comment]
@Gauravpadam I updated all the changes as per your latest feedback. Please review it onnce

**Comment by Gauravpadam** [comment]
Thanks, I'll be happy to lgtm if either:

- As per your suggestion in [#44527(comment)](https://github.com/kubernetes/website/pull/44527#issuecomment-1869640856), You make the css external in a new PR and keep this only for html changes
- You add a new commit migrating the internal css to `/assets/custom.scss`, i.e. our external css assets

**Comment by Subhajit009iitr** [comment]
> Thanks, I'll be happy to lgtm if either:
> 
> * As per your suggestion in [#44527(comment)](https://github.com/kubernetes/website/pull/44527#issuecomment-1869640856), You make the css external in a new PR and keep this only for html changes
> * You add a new commit migrating the internal css to `/assets/custom.scss`, i.e. our external css assets

I have moved the CSS component to `/assets/_custom.scss` as you instructed.

**Review by Gauravpadam — COMMENTED** [review]
/lgtm

Thanks for the PR @Subhajit009iitr!

**Comment by Gauravpadam** [comment]
Some more [web development](https://github.com/kubernetes/website/issues?q=is%3Aissue+is%3Aopen+label%3Aarea%2Fweb-development+label%3Atriage%2Faccepted%2Cpriority%2Fimportahttps%3A%2F%2Fgithub.com%2Fkubernetes%2Fwebsite%2Fpulsent-longterm+sort%3Acreated-asc++-linked%3Apr+) issues that could be worked on - only if you'd want to.

**Comment by Subhajit009iitr** [comment]
> Some more [web development](https://github.com/kubernetes/website/issues?q=is%3Aissue+is%3Aopen+label%3Aarea%2Fweb-development+label%3Atriage%2Faccepted%2Cpriority%2Fimportahttps%3A%2F%2Fgithub.com%2Fkubernetes%2Fwebsite%2Fpulsent-longterm+sort%3Acreated-asc++-linked%3Apr+) issues that could be worked on - only if you'd want to.

Ya I would Love to! I will look into them. Thanks

**Comment by divya-mohan0209** [comment]
Hello @Subhajit009iitr : Please could you add an appropriate description of the changes you've made in this PR and why you think these changes are necessary? This would help approvers understand your POV.

Explicit hold on the PR till the description is added.
/hold

**Comment by Subhajit009iitr** [comment]
> Hello @Subhajit009iitr : Please could you add an appropriate description of the changes you've made in this PR and why you think these changes are necessary? This would help approvers understand your POV.
> 
> Explicit hold on the PR till the description is added. /hold

I have updated the description as instructed here
https://github.com/kubernetes/website/pull/44527#issue-2056201529

**Comment by Subhajit009iitr** [comment]
@divya-mohan0209 Is the description fine?

**Comment by divya-mohan0209** [comment]
Thank you @Subhajit009iitr for getting to this so promptly!
/unhold
/approve

**Pull request merged**